lib/new_relic/agent/error_filter.rb

```diff
@@ -0,0 +1,153 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+module NewRelic
+  module Agent
+
+    # Handles loading of ignored and expected errors from the agent configuration, and
+    # determining at runtime whether an exception is ignored or expected.
+    class ErrorFilter
+
+      def initialize
+        @ignore_errors, @ignore_classes, @expected_classes = [], [], []
+        @ignore_messages, @expected_messages = {}, {}
+        @ignore_status_codes, @expected_status_codes = [], []
+      end
+
+      def load_all
+        %i(
+          ignore_errors ignore_classes ignore_messages ignore_status_codes
+          expected_classes expected_messages expected_status_codes
+        ).each { |setting| load_from_config(setting) }
+      end
+
+      def load_from_config(setting, value = nil)
+        errors = nil
+        new_value = value || fetch_agent_config(setting.to_sym)
+        case setting.to_sym
+        when :ignore_errors  # Deprecated; only use if ignore_classes isn't present
+          errors = @ignore_errors = new_value.to_s.split(',').map!(&:strip)
+        when :ignore_classes
+          errors = @ignore_classes = new_value || []
+        when :ignore_messages
+          errors = @ignore_messages = new_value || {}
+        when :ignore_status_codes
+          errors = @ignore_status_codes = parse_status_codes(new_value) || []
+        when :expected_classes
+          errors = @expected_classes = new_value || []
+        when :expected_messages
+          errors = @expected_messages = new_value || {}
+        when :expected_status_codes
+          errors = @expected_status_codes = parse_status_codes(new_value) || []
+        end
+        log_filter(setting, errors)
+      end
+
+      # Define #ignored? and #expected? in this way so that any given exception
+      # cannot be both ignored and expected when using type_for_exception.
+      # Ignoring takes priority.
```

Comment:
This might be a leftover comment from an earlier version of the code. Ignoring should naturally take precedence since the it's the first thing the error collector checks. I'll make a note to remove this comment.

Reply:
Ahhh okay that makes sense. I did notice in the main error flow that it's checking ignore first, but I got confused by the comment. ty!